docs/cryptodoc/src/05_08_ml_dsa.rst

```diff
@@ -0,0 +1,443 @@
+classes are the internal representation of the ML-DSA key pair in expandedd form.
+
+Additionally, the :srcref:`Dilithium_Keypair_Codec
+<src/lib/pubkey/dilithium/dilithium_common/dilithium_keys.h:23|Dilithium_Keypair_Codec>`
+serves as a customization point for the key encoding and decoding functions that
+differ between ML-DSA (:math:`\xi` only) and Dilithium (round 3) (partially
+expanded key format as specified in [FIPS-204]_). By *always* expanding the
+private key from the secret seed :math:`\xi`, sanity checks during decoding of
+the key pair can be omitted.
+
+.. _pubkey/ml_dsa/ml_dsa_impl:
+
+ML-DSA Specifics
+^^^^^^^^^^^^^^^^
+
+This module provides concrete ML-DSA specific implementations for the
+customization points outlined in :ref:`pubkey/ml_dsa/primitives` and
+:ref:`pubkey/ml_dsa/keys`. Namely:
+
+  * :srcref:`ML_DSA_Expanding_Keypair_Codec <src/lib/pubkey/dilithium/ml_dsa/ml_dsa_impl.h:21|ML_DSA_Expanding_Keypair_Codec>`
+    Implements encoding and decoding of ML-DSA private keys by serializing the
+    private seed :math:`\xi` and/or expanding the deserialized seed into the
+    private key representation outlined in :ref:`pubkey/ml_dsa/keys`.
+  * :srcref:`ML_DSA_MessageHash <src/lib/pubkey/dilithium/ml_dsa/ml_dsa_impl.h:28|ML_DSA_MessageHash>`
+    Implements the transformation of the user-provided message :math:`M` into
+    the message representation :math:`\mu`. This includes the incorporation of
+    the domain separations outlined in [FIPS-204]_ Sectoin 5.2 Algorithm 2.
+  * :srcref:`ML_DSA_Symmetric_Primitives
+    <src/lib/pubkey/dilithium/ml_dsa/ml_dsa_impl.h:54|ML_DSA_Symmetric_Primitives>`
+    Implements the ML-DSA specific symmetric primitives based on the specified
+    SHAKE-based XOFs, the optional hedged randomization of :math:`H(K \| rnd \| \mu)`,
+    and the domain separator for expanding :math:`\rho`, :math:`\rho'`, and
+    :math:`K` from the private seed :math:`\xi`.
+
+.. _pubkey/ml_dsa/ml_dsa_api:
+
+Public API
+^^^^^^^^^^
+
+The :srcref:`Dilithium_PublicKey
+<src/lib/pubkey/dilithium/dilithium_common/dilithium.h:67|Dilithium_PublicKey>`
+and :srcref:`Dilithium_PrivateKey
+<src/lib/pubkey/dilithium/dilithium_common/dilithium.h:115|Dilithium_PrivateKey>`
+classes serve as Botan's public API for public and private ML-DSA keys,
+respectively. The :srcref:`DilithiumMode
+<src/lib/pubkey/dilithium/dilithium_common/dilithium.h:21|DilithiumMode>` class
+is used to select the desired parameter set.
+
+New applications that do not rely on the pre-standard Dilithium round 3
+implementations are strongly advised to use the type aliases for ML-DSA defined
+in :srcref:`[src/lib/pubkey/dilithium/ml_dsa]/ml_dsa.h`.
+
+
+.. _pubkey/ml_dsa/kyber_compat:
+
+Dilithium Compatibility
+-----------------------
+
+The final ML-DSA standard is not compatible with the round 3 submission of
+Dilithium. Botan did provide support for Dilithium and Dilithium-AES as
```

Comment:
```suggestion
Dilithium. Botan has provided support for Dilithium and Dilithium-AES as
```
(says Grammarly)

Reply:
I think it is right. "did provide" would be correct, if the support was removed in the past.